core/services/application.go

```diff
@@ -49,13 +50,18 @@ func NewApplication(config *orm.Config, onConnectCallbacks ...func(Application))
 	store := store.NewStore(config)
 	config.SetRuntimeStore(store.ORM)
 
-	jobSubscriber := NewJobSubscriber(store)
-	pendingConnectionResumer := newPendingConnectionResumer(store)
+	jobExecutor := NewJobExecutor(store)
+	jobRunner := NewJobRunner(jobExecutor)
+	jobManager := NewJobManager(jobRunner, config, store.ORM, store.TxManager, store.Clock)
```

Comment:
Split these out as I'm trying to avoid "god objects" and think we should consider a DI framework here. _Only inject what is needed_

Reply:
So I went with:

  * JobManager -> RunManager
  * JobRunner -> RunQueue
  * JobExecutor -> RunExecutor

My main reasoning for differing from your suggestions were:

  1. The TaskExecutor also executes Runs so renaming it was a little inaccurate
  2. The RunQueue's only difference from just starting a go routine for every Run is that it queues run requests, it defers execution to the RunExecutor, so this felt like a better name and made the name RunExecutor available for (1)